corehq/apps/api/odata/serializers.py

```diff
@@ -85,6 +85,13 @@ def serialize_documents_using_config(documents, config, table_id):
 
         data = []
         for row_number, document in enumerate(documents):
+            if 'location_id' in document and 'location_id' in document['case_json']:
+                if document['location_id'] is None:
```

Comment:
I don't think this could ever happen unless both `case.location_id` and `case.case_json['location_id']` are both None (which I also don't think is possible since case properties can't be set to `None`).

It would be great to have some tests for this.

Reply:
> Is there a reason we can't put the value into the `location_id` case JSON property in addition to setting `case.location_id`?

The scenario I thought might occur is that a case may have a legitimate `location_id` (corresponding to a Location in the project) assigned to it but in a followup form (can supply point cases have those?) have a separate case property defined also as `location_id` with a user input as its value. If this doesn't seem like a possible scenario I think we can just have it assigned in both places without problem but the possibility that one might overwrite the other is why I tried to separate them.

Reply:
With the changes in this branch at least, I was seeing that `location_id` as defined as a case property in `case_json` was coming up blank in Power BI reports, because I believe it was being overwritten by `case.location_id`, which if a case isn't assigned to a location is usually empty. This check does fix the issue of the missing value for `location_id` so there are scenarios like this. 

Also, I agree tests would be great! Working on them now.